deepmd/pt/model/descriptor/se_atten.py

```diff
@@ -478,47 +509,114 @@ def forward(
-            gg = gg_s * gg_t + gg_s
-        else:
-            raise NotImplementedError
+                # nfnl x nnei x ng
+                gg_s = self.filter_layers.networks[0](ss)
+                assert self.filter_layers_strip is not None
+                if not self.type_one_side:
+                    # nfnl x nnei x (tebd_dim * 2)
+                    tt = torch.concat([nlist_tebd, atype_tebd], dim=2)  # dynamic, index
+                else:
+                    # nfnl x nnei x tebd_dim
+                    tt = nlist_tebd
+                # nfnl x nnei x ng
+                gg_t = self.filter_layers_strip.networks[0](tt)
+                if self.smooth:
+                    gg_t = gg_t * sw.reshape(-1, self.nnei, 1)
+                # nfnl x nnei x ng
+                gg = gg_s * gg_t + gg_s
+                input_r = torch.nn.functional.normalize(
+                    rr.reshape(-1, self.nnei, 4)[:, :, 1:4], dim=-1
+                )
+                gg = self.dpa1_attention(
+                    gg, nlist_mask, input_r=input_r, sw=sw
+                )  # shape is [nframes*nloc, self.neei, out_size]
+                # nfnl x 4 x ng
+                xyz_scatter = torch.matmul(rr.permute(0, 2, 1), gg)
 
-        input_r = torch.nn.functional.normalize(
-            rr.reshape(-1, self.nnei, 4)[:, :, 1:4], dim=-1
-        )
-        gg = self.dpa1_attention(
-            gg, nlist_mask, input_r=input_r, sw=sw
-        )  # shape is [nframes*nloc, self.neei, out_size]
-        # nfnl x 4 x ng
-        xyz_scatter = torch.matmul(rr.permute(0, 2, 1), gg)
         xyz_scatter = xyz_scatter / self.nnei
         xyz_scatter_1 = xyz_scatter.permute(0, 2, 1)
         rot_mat = xyz_scatter_1[:, :, 1:4]
         xyz_scatter_2 = xyz_scatter[:, :, 0 : self.axis_neuron]
         result = torch.matmul(
             xyz_scatter_1, xyz_scatter_2
         )  # shape is [nframes*nloc, self.filter_neuron[-1], self.axis_neuron]
-        return (
-            result.view(nframes, nloc, self.filter_neuron[-1] * self.axis_neuron),
-            gg.view(nframes, nloc, self.nnei, self.filter_neuron[-1]),
-            dmatrix.view(nframes, nloc, self.nnei, 4)[..., 1:],
-            rot_mat.view(nframes, nloc, self.filter_neuron[-1], 3),
-            sw,
-        )
+
+        if not self.compress:
+            return (
+                result.view(-1, nloc, self.filter_neuron[-1] * self.axis_neuron),
+                gg.view(-1, nloc, self.nnei, self.filter_neuron[-1]),
+                dmatrix.view(-1, nloc, self.nnei, 4)[..., 1:],
+                rot_mat.view(-1, nloc, self.filter_neuron[-1], 3),
+                sw,
+            )
+        else:
+            return (
+                result.view(-1, nloc, self.filter_neuron[-1] * self.axis_neuron),
+                None,
+                dmatrix.view(-1, nloc, self.nnei, 4)[..., 1:],
+                rot_mat.view(-1, nloc, self.filter_neuron[-1], 3),
+                sw,
+            )
```

Comment:
should use `nframes` instead of `-1`.

may written as

```py
gg.view(nframes, nloc, self.nnei, self.filter_neuron[-1]) if not self.compress else None, 
```